### ISSUE TITLE
[BUG]: True Crime: New York: control issues

### ISSUE THREAD

**Issue by Shoegzer** [PCSX2/pcsx2#7584]
### Describe the Bug

Frequently, when the analog stick is pushed up to move the character, character actually moves up and to the left.  This is not an issue in other games, for example in Scarface, pressing up on the analog stick results in the character moving up as expected.

Test syetem:
* OS: Linux Mint 21 / kernel 5.15
* Gamepad: Sony DualSense (DS5) pad

### Reproduction Steps

Start a new game, at the very beginning (Prologue), you will be able to move the character and observe the issue.

### Expected Behavior

Character moves as expected (up results in moving up).

### PCSX2 Revision

Commit 354951f1d62e33bc94fcf650f993f42742a5e4f3

### Operating System

Linux (64bit) - Specify Distro Below

### If Linux - Specify Distro

Linux Mint 21

**Comment by Shoegzer** [comment]
Update: Tested with keyboard controls as well, but still encounter the control issue, so it doesn't seem tied to an input method.  Would appreciate someone testing this on Windows to rule out an SDL2 issue as well. In gameplay I observed some of the other characters (perps) running into left-side walls instead of straight ahead while chasing them, so perhaps the control problem is symptomatic of a larger issue with the emulation.

Also, the issue only seems to occur when the camera is facing a certain direction. In other words, sometimes the character moves up when the left analog stick is pushed up, but when the camera is moved to a new direction, the character moves up and to the left when the left analog stick is pushed up.  Strange behavior.

I was going to test this against changes to EE and VU rounding/clamping but system settings seem to be gone with the latest commits (?) and I couldn't find any reasons for why they were removed, so I'll test if/when they return. @refractionpcsx2 are these now exposed as a hidden setting or am I just blind and totally missing them?
